File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    left, right = 0, 0
    maxDiff = 0
    length = len(prices)
    while right != length:
      maxDiff = max(maxDiff, prices[right] - prices[left])
      if prices[left] > prices[right]:
        left = right
      else:
        right += 1
        
    return maxDiff

File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.cpp

class Solution {
public:
    bool isValidBST(TreeNode* root) {
      return
        helper(root->left, root->val, std::numeric_limits<int>::min()) &&
        helper(root->right, std::numeric_limits<int>::max(), root->val);
    }

    bool helper(TreeNode* node, int biggest, int smallest) {
      if (!node) return true;

      if (node->val >= biggest || node->val <= smallest) return false;

      bool leftValid;
      if (!node->left) leftValid = true;
      else leftValid = helper(node->left, node->val, smallest);

      bool rightValid;
      if (!node->right) rightValid = true;
      else rightValid = helper(node->right, biggest, node->val);

      return leftValid && rightValid;
    }
};
